src/hooks/useOrganizations.tsx

```diff
@@ -0,0 +1,42 @@
+import { personalOctokit } from 'bot/octokit'
+import { useSession } from 'next-auth/react'
+import router from 'next/router'
+import { useEffect, useState } from 'react'
+
+const getOrganizationsData = async (accessToken: string) => {
+  try {
+    const octokit = personalOctokit(accessToken)
+    const data = await octokit.rest.orgs.listForAuthenticatedUser()
+    return data.data
+  } catch (error) {
+    return null
+  }
+}
+
+export const useOrgsData = () => {
+  const session = useSession()
+  const accessToken = session.data?.user.accessToken
+
+  const [organizationsData, setOrganizationsData] = useState<Awaited<
+    ReturnType<typeof getOrganizationsData>
+  > | null>(null)
+
+  useEffect(() => {
+    if (!accessToken) {
+      return
+    }
+
+    getOrganizationsData(accessToken).then((orgs) => {
+      // redirect to org page if user is only in one org
+      if (orgs && orgs.length === 1) {
+        router.push(`/${orgs[0].login}`)
```

Comment:
I think we may want to perform this navigation action on a page rather than a hook. I got an error for trying to execute this function server side

<img width="1728" alt="Screenshot 2024-06-05 at 12 47 06 PM" src="https://github.com/github-community-projects/internal-contribution-forks/assets/24923406/419ed9b2-4cb4-4b87-a7cb-9dc02f5d13d8">

Reply:
I moved it to the home page, but then realized it would redirect every time that the user would try to go to the home page, so I moved it to the login page.